tests/php/ORM/DataListTest.php

```diff
@@ -4,34 +4,58 @@
 
 use InvalidArgumentException;
 use SilverStripe\Core\Convert;
-use SilverStripe\Core\Injector\InjectorNotFoundException;
+use SilverStripe\Dev\SapphireTest;
 use SilverStripe\ORM\DataList;
 use SilverStripe\ORM\DataQuery;
 use SilverStripe\ORM\DB;
 use SilverStripe\ORM\Filterable;
 use SilverStripe\ORM\Filters\ExactMatchFilter;
-use SilverStripe\Dev\SapphireTest;
-use SilverStripe\ORM\Tests\DataObjectTest\Fixture;
 use SilverStripe\ORM\Tests\DataObjectTest\Bracket;
 use SilverStripe\ORM\Tests\DataObjectTest\EquipmentCompany;
 use SilverStripe\ORM\Tests\DataObjectTest\Fan;
+use SilverStripe\ORM\Tests\DataObjectTest\Fixture;
+use SilverStripe\ORM\Tests\DataObjectTest\House;
+use SilverStripe\ORM\Tests\DataObjectTest\HouseVisit;
 use SilverStripe\ORM\Tests\DataObjectTest\Player;
+use SilverStripe\ORM\Tests\DataObjectTest\Roof;
 use SilverStripe\ORM\Tests\DataObjectTest\Sortable;
+use SilverStripe\ORM\Tests\DataObjectTest\Staff;
 use SilverStripe\ORM\Tests\DataObjectTest\SubTeam;
 use SilverStripe\ORM\Tests\DataObjectTest\Team;
 use SilverStripe\ORM\Tests\DataObjectTest\TeamComment;
 use SilverStripe\ORM\Tests\DataObjectTest\ValidatedObject;
-use SilverStripe\ORM\Tests\DataObjectTest\Staff;
+use SilverStripe\ORM\Tests\DataObjectTest\Visitor;
+use SilverStripe\ORM\Tests\DataObjectTest\WoodenRoof;
+use SilverStripe\ORM\Tests\ManyManyListTest\Category;
+use SilverStripe\ORM\Tests\ManyManyListTest\Product;
+use SilverStripe\Versioned\Versioned;
 
 /**
  * @skipUpgrade
  */
 class DataListTest extends SapphireTest
 {
-
     // Borrow the model from DataObjectTest
     protected static $fixture_file = 'DataObjectTest.yml';
 
+    /**
+     * @var array
+     */
+    protected static $required_extensions = [
+        House::class => [
+            Versioned::class,
+        ],
+        HouseVisit::class => [
+            Versioned::class,
+        ],
+        Visitor::class => [
+            Versioned::class,
+        ],
+        Roof::class => [
+            Versioned::class,
```

Comment:
Maybe these tests belong in the versioned module? If we do keep them here then these tests will fail without versioned installed (logically, since the class not existing as a string literal is translucent but acceptable) - we may want to mark these tests as skipped if the module is not installed. It could even become its own test class with this in the setup method.